### PR TITLE
feat: Add `chat/send-message` with slack and discord

### PR DESCRIPTION
**Pull request by martinalbert**
## Description
Adds new profile `chat/send-message` with slack and discord maps.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: grid/chat/send-message/profile.supr [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+"""
+Send Message
+Sends message to one destination
+"""
+usecase SendMessage {
+  input {
+    """
+    Destination
+    Identification of destination to retrieve messages from
```
Comment (jnv) [inline grid/chat/send-message/profile.supr:12]: Copy pasta? :wink: 

```suggestion
    Channel to send the message to
```

File: grid/chat/send-message/profile.supr [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    """
+    destination! string!
+    
+    """
+    Text
+    Content of the message
```
Comment (jnv) [inline grid/chat/send-message/profile.supr:18]:
```suggestion
    Contents of the message
```

File: grid/chat/send-message/maps/send-message.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
+/* eslint-disable jest/no-export */
+
+import { RecordingProcessOptions, SuperfaceTest } from '@superfaceai/testing';
+
+export const sendMessageTest = (
+  provider: string,
+  destination: string[],
```
Comment (jnv) [inline grid/chat/send-message/maps/send-message.ts:7]: TBH this API doesn't tell anything about the fact that the second destination is supposed to be non-existent. As a quick workaround, you can define this parameter as a tuple, but consider better function interfaces in the future.

**Review by jnv — CHANGES_REQUESTED** [review]

**Comment by jnv** [comment]
@martinalbert What is the status here? Is this ready?

**Comment by martinalbert** [comment]
@jnv yes it is ready.

File: grid/chat/send-message/maps/discord.suma [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+  set if (!isNaN(resetTimestamp)) {
+    details.resetTimestamp = resetTimestamp * 1000
+  }
+
+  set if (!isNaN(retryAfter)) {
+    details.retryAfter = retryAfter
+  }
+
+  set if (details.remainingRequests && details.totalRequests) {
```
Comment (janhalama) [inline grid/chat/send-message/maps/discord.suma:79]:
```suggestion
  set if (details.remainingRequests !== undefined && details.totalRequests) {
```

File: grid/chat/send-message/maps/slack.suma [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+  set if (!isNaN(resetAfter)) {
+    details.resetAfter = resetAfter
+    details.resetTimestamp = Date.now() + (resetAfter * 1000)
+  }
+
+  set if (!isNaN(retryAfter)) {
+    details.retryAfter = retryAfter
+  }
+
+  set if (details.remainingRequests && details.totalRequests) {
```
Comment (janhalama) [inline grid/chat/send-message/maps/slack.suma:81]:
```suggestion
  set if (details.remainingRequests !== undefined && details.totalRequests) {
```

**Comment by martinalbert** [comment]
@janhalama thanks, I added your suggestions.

**Review by janhalama — APPROVED** [review]
LGTM

**Review by jnv — APPROVED** [review]
:shipit:

**Pull request merged**